### PR TITLE
xtensa: Add mt8195_adsp toolchain

### PR DESCRIPTION
**Pull request by andyross**
Yet another Xtensa SOC for the SDK.  Mostly the same as the other LX cores, but different enough to not be able to share code. (No MULUH instruction on this model, which breaks on compiler-generated 64 bit multiplies).

[Submitting this blindly to see if a usable SDK pops out magically, as per the docs.  Hopefully there's help available when something breaks.  Isn't there a Dockerfile or something I can use to do a build somewhere?]

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (dcpleung) [inline .github/workflows/ci.yml:1539]: Will need to update this after the board config has been merged.

Comment (andyross) [reply .github/workflows/ci.yml:1539]: Oops, yeah, that's a cut/paste error for sure.  Will fix.

**Comment by dbaluta** [comment]
@andyross This looks good to me. 

FYI: https://github.com/zephyrproject-rtos/sdk-ng/pull/187/commits here is how @lgirdwood did the changes for Inte/NXP.

File: .github/workflows/ci.yml
Comment (sylvioalves) [inline .github/workflows/ci.yml:167]: For the sake of alignment, would you add a line break here? `build_target....`

**Comment by dbaluta** [comment]
@andyross  I'm also interested about the toolchain automagically being built by the CI.

Anyone can shed some light on this? :)

**Review by dbaluta — APPROVED** [review]

**Comment by nashif** [comment]
> But the new toolchain isn't there. It only built the old ones. I'm pretty sure it was added to all the places needed (basically I cloned the settings for intel_tgl_adsp, everywhere it's mentioned there's a line for mtk8195_adsp). Is the issue maybe that CI runs the workflow from the release and not the code in the PR? How do I fix this?

this is because of the action mode (pull_request_target). security blah blah

**Review by lgirdwood — APPROVED** [review]

**Comment by dbaluta** [comment]
@andyross  this needs rebase.

**Comment by andyross** [comment]
Rebased & fixed up, should hopefully work.

FWIW: I noticed while mucking with this a while back that there appears to be (vestigial?) support for multi-target gcc xtensa builds upstream.  Has anyone every looked at this?  Basically there's an "xtensa-dynconfig.c" file in upstream binutils and gcc, and it appears to look at a "XTENSA_GNU_CONFIG" environment variable, which it appears to interpret as a shared library that it dlopens to get the various tables normally built out of the overlay.

Has anyone else every noticed this or tried it?   This would save HUGE amounts of space in the SDK if we could make it work.  It looks semi-abandoned, but not complicated.

**Comment by andyross** [comment]
(FWIW: also need to get the mt8195 HAL additions updated too, I think there were comments.  But the board PR itself is currently buried and horribly entangled with the call0 work, so that's going to take a bit to unravel)

File: overlays/xtensa_mtk_mt8195/gdb/gdb/xtensa-config.c
Comment (dcpleung) [inline overlays/xtensa_mtk_mt8195/gdb/gdb/xtensa-config.c:504]: Need to add:
```
xtensa_gdbarch_tdep xtensa_tdep (rmap);
```
or else it won't build.

**Review by dcpleung — CHANGES_REQUESTED** [review]

File: overlays/xtensa_mtk_mt8195/gdb/gdb/xtensa-config.c
Comment (andyross) [reply overlays/xtensa_mtk_mt8195/gdb/gdb/xtensa-config.c:504]: Sorry, I'm lost on that requirement.  Where exactly do I put that and why?  :)

This is just the upstream file from the Cadence SDK (with some path munging via that make-overlay.sh script at foss-xtensa).  Is this something they've forgotten or that we need to maintain and didn't document?  Or am I looking at the wrong place?

Comment (andyross) [reply overlays/xtensa_mtk_mt8195/gdb/gdb/xtensa-config.c]: Ah, OK.  Skew vs. upstream gdb APIs.  Found the other commits in the tree.

We really need to write a "overlay importer" layer or something.  To add to the list with "make dynamic configs work" I guess.  Managing this is sort of a mess for someone doing it the first time, and we're juggling a ton of these toolchains these days.

File: .github/workflows/ci.yml
Comment (nashif) [inline .github/workflows/ci.yml:1554]: Do we have a PR introducing this? We need to have this ready somehow and merge it or put it in the sdk branch to get this test to build, otherwise CI will fail.

Comment (andyross) [reply .github/workflows/ci.yml:1554]: PR for the board is at https://github.com/zephyrproject-rtos/zephyr/pull/61197 but I wasn't pushing it as there are still some review comments and IMHO it makes more sense to wait for the call0 work to land first as then it can launch with working userspace (it's got an MPU).

Isn't that a chicken and egg problem?  I assumed we needed the SDK patch before the board could merge too?  Maybe I should remove these lines and merge an update later?

**Review by dcpleung — APPROVED** [review]

File: .github/workflows/ci.yml
Comment (nashif) [reply .github/workflows/ci.yml:1554]: it is a chicken and egg problem, but when we test the SDK in CI, we pull this from a branch (topic-sdk-dev), so we can put this PR for building purposes and while we wait for it to get merged in zephyr main.

**Pull request merged**
